dep0/lib/05_llvmgen/src/gen.cpp

```diff
@@ -360,28 +363,26 @@ llvm::Attribute::AttrKind get_sign_ext_attribute(local_context_t const& local, t
                         {
                             return x.sign == ast::sign_t::signed_v ? llvm::Attribute::SExt : llvm::Attribute::ZExt;
                         });
-                },
-                [&] (typecheck::type_t const& type)
-                {
-                    return get_sign_ext_attribute(local, type);
                 });
         },
-        [] (typecheck::type_t::arr_t const&) { return llvm::Attribute::None; });
+        [] (typecheck::expr_t::app_t const&) { return llvm::Attribute::None; },
+        [] (typecheck::expr_t::abs_t const&) { return llvm::Attribute::None; },
+        [] (typecheck::expr_t::pi_t const&) { return llvm::Attribute::None; });
 }
 
 llvm::FunctionType* gen_func_type(
     global_context_t& global,
     local_context_t const& local,
-    llvm_func_proto_t const& proto)
+    typecheck::expr_t::pi_t const& proto)
```

Comment:
could potentially have a `first_order_function_type_t` wrapping around the `pi_t` as proof that the wrapped `pi_t` is indeed a 1st order function type

Reply:
maybe can reinstante `llvm_func_proto_t` and make it mean this